hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/db/TimePartitionServiceImpl.java

```diff
@@ -21,33 +21,67 @@
 import static com.hedera.mirror.importer.config.CacheConfiguration.CACHE_TIME_PARTITION_OVERLAP;
 
 import com.google.common.collect.Range;
+import com.hedera.mirror.importer.util.Utility;
 import jakarta.inject.Named;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 import lombok.CustomLog;
-import lombok.RequiredArgsConstructor;
-import org.springframework.cache.annotation.Cacheable;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.cache.Cache;
+import org.springframework.cache.CacheManager;
 import org.springframework.jdbc.core.JdbcTemplate;
 import org.springframework.jdbc.core.RowMapper;
 
 @CustomLog
 @Named
-@RequiredArgsConstructor
 public class TimePartitionServiceImpl implements TimePartitionService {
 
+    private final Cache cacheTimePartitionOverlap;
+    private final Cache cacheTimePartition;
+    private final JdbcTemplate jdbcTemplate;
     private static final String GET_TIME_PARTITIONS_SQL = "select * from mirror_node_time_partitions where parent = ?";
     private static final RowMapper<TimePartition> ROW_MAPPER = (rs, rowNum) -> TimePartition.builder()
             .name(rs.getString("name"))
             .parent(rs.getString("parent"))
             .timestampRange(Range.closedOpen(rs.getLong("from_timestamp"), rs.getLong("to_timestamp")))
             .build();
 
-    private final JdbcTemplate jdbcTemplate;
+    TimePartitionServiceImpl(
+            @Qualifier(CACHE_TIME_PARTITION_OVERLAP) CacheManager cacheManagerOverlapTimePartition,
+            @Qualifier(CACHE_TIME_PARTITION) CacheManager cacheManagerTimePartition,
+            JdbcTemplate jdbcTemplate) {
+        this.cacheTimePartitionOverlap = cacheManagerOverlapTimePartition.getCache(CACHE_NAME);
+        this.cacheTimePartition = cacheManagerTimePartition.getCache(CACHE_NAME);
+        this.jdbcTemplate = jdbcTemplate;
+    }
 
-    @Cacheable(cacheManager = CACHE_TIME_PARTITION_OVERLAP, cacheNames = CACHE_NAME)
     @Override
     public List<TimePartition> getOverlappingTimePartitions(String tableName, long fromTimestamp, long toTimestamp) {
+        String cacheKey = StringUtils.joinWith("-", tableName, fromTimestamp, toTimestamp);
```

Comment:
IntelliJ profiler shows `StringUtils.joinWith` is the hotspot, this should be much faster
```suggestion
        String cacheKey = tableName + "-" +  fromTimestamp + "-" + toTimestamp;
```

Reply:
Thanks! I have updated the concatenation.